---
title: "Introduction to _CommunityAMARETTO_"
author:
  - name: Mohsen Nabian
    affiliation: Brigham and Women's Hospital, Harvard Medical School, Broad Institute of MIT and Harvard, Boston and Cambridge, MA, USA
  - name: Celine Everaert
    affiliation: Brigham and Women's Hospital, Harvard Medical School, Broad Institute of MIT and Harvard, Boston and Cambridge, MA, USA
  - name: Artur Manukiyan
    affiliation: Brigham and Women's Hospital, Harvard Medical School, Broad Institute of MIT and Harvard, Boston and Cambridge, MA, USA
  - name: Shaimaa Bakr
    affiliation: Stanford Center for Biomedical Informatics Research (BMIR), Department of Medicine and Biomedical Data Science, 1265 Welch Rd, Stanford, CA, USA
  - name: Jishu Xu
    affiliation: RUSH University Medical Center, Chicago, IL, USA; Cell Circuits Program, Broad Institute of MIT and Harvard, MA, USA
  - name: Mikel Hernaez
    affiliation: CIMA, University of Navarra,Spain; Carl R. Woese Institute for genomic Biology, University of Illinois, Urbana-Champaign, IL, USA
  - name: Vincent Carey
    affiliation: Brigham and Women's Hospital, Harvard Medical School, Broad Institute of MIT and Harvard, Boston and Cambridge, MA, USA
    footnote: Corresponding Author
  - name: Olivier Gevaert
    affiliation: Stanford Center for Biomedical Informatics Research (BMIR), Department of Medicine and Biomedical Data Science, 1265 Welch Rd, Stanford, CA, USA
    email: olivier.gevaert@stanford.edu
    footnote: Corresponding Author
  - name: Nathalie Pochet
    affiliation: Brigham and Women's Hospital, Harvard Medical School, Broad Institute of MIT and Harvard, Boston and Cambridge, MA, USA
    email: npochet@broadinstitute.org

date: "`r BiocStyle::doc_date()`"
output:
    BiocStyle::html_document

abstract: |
    The goal of the CommunityAMARETTO algorithm is to identify cell circuits
    and their drivers that are shared and distinct across biological systems.
    Specifically, CommunityAMARETTO takes as input multiple regulatory
    networks inferred using the AMARETTO algorithm [1] that are based
    on multi-omics and clinical/imaging data fusion. Next, CommunityAMARETTO
    learns communities or subnetworks, in particular, regulatory modules
    comprising of cell circuits and their drivers, that are shared and
    distinct across multiple regulatory networks derived from 
    multiple cohorts, diseases, or biological systems more generally,
    using the Girvan-Newman "edge betweenness community detection"
    algorithm (Girvan and Newman, Physical Review E. 2004).

package: Report issues on [https://github.com/broadinstitute/CommunityAMARETTO/] (https://github.com/broadinstitute/CommunityAMARETTO/)
vignette: |
    %\VignetteIndexEntry{Introduction to _CommunityAMARETTO_}
    %\VignetteEngine{knitr::rmarkdown}
    %\VignetteEncoding{UTF-8}
    \usepackage[utf8]{BiocStyle}
---

# Introduction :

Uncovering gene regulatory mechanisms underlying diseases and cancer
systems have long time been sought by many researchers. 
This led to the development of many novel computational algorithms for
regulatory network inference using multiomics such as genetics,
epigenetics and transcriptomics.

These algorithms are designed to infer system-specific gene regulatory
network, leading to the understanding of underlying regulatory mechanisms
of the system and facilitating biomarker discovery.

It is also believed that the integration of regulatory networks across
multiple systems provides key information about cross-systems shared
and distinct mechanisms. To this end, we developed *Community-AMARETTO*
algorithm to integrate multiple regulatory networks inferred by the
AMARETTO algorithm [1]. Specifically, Community-AMARETTO algorithm consists of
1) constructing a master network composed of multiple regulatory networks
followed by 2) detecting groups (communities) of circuits that are shared
across systems as well as highliting circuits that are system-specific
and distinct. 

As general guidelines, here we provided instruction for running
Community-AMARETTO on a sample case study of liver cancer.
While here we studied liver-specific data, any cohort/dataset analyzed
by AMARETTO can be used for the Community-AMARETTO analysis. 

Our sample case study consists of three biological systems of
1) Liver Cancer, 2) Hepatite C virus-infected liver cancer and 
3) Hepatite B virus-infected liver cancer, each inferred by the AMARETTO 
algorithm for gene regulatory network inference.

In this note, we show how to integrate these networks through
Community-AMARETTO algorithm, perform downstream analysis such as
functional enrichment of the detected communities and to generate interactive
HTML-report from the analysis results.

# Installing Package

The following code script shows how to install the package. The package is
accessible via Bioconductor; Alternatively, to access the newest features
of the pacckage, you may install the development branch of github
as shown bellow. 

```{r AMARETTO_Run, message= FALSE,warning=FALSE,results='hide'}
library(devtools)
#install.packages("/Users/mohsennabian/Documents/GitHub/CommunityAMARETTO",
#repos = NULL, type = "source")
install_github("broadinstitute/CommunityAMARETTO",
                ref='develop',
                force = 'TRUE')
library(CommunityAMARETTO)
```


# Loading data

CommunityAMARETTO is a network approach developed to integrate multiple
regulatory networks specifically networks inferred by AMARETTO algorithm.
Thus,the required input data for the CommunityAMARETTO algorithm are
AMARETTO-result objects (at least two); where each AMARETTO-result object
is the output of an AMARETTO run on multi-omics data and contains 
the information for the inferred regulatory network parameters.

Here as a sample case study, we provided AMARETTO-result objects 
(outputs of AMARETTO runs) for 3 liver-related datasets :
1) TCGA_LIHC_data contains AMARETTO network parameters infered on
TCGA-LIHC multiomics data [2] 2) scHCV_data contains AMARETTO network
parameters infered on transcriptomics of single cell HCV virus infected
liver samples [3], and similarly 3) scHBV_data contains AMARETTO network
parameters infered on transcriptomics of single cell HBV virus infected
liver samples [3]. These sample AMARETTO-result objects can be accessed
through the CommunityAMARETTO package as shown below, and are derived 
from a small portion of the real data, therefore are for representation
only. 


```{r AMARETTO_Run1, message= FALSE,warning=FALSE,results='hide'}
#Load data
TCGA_LIHC_data<-CommunityAMARETTO::TCGA_LIHC_data
scHCV_data<-CommunityAMARETTO::scHCV_data
scHBV_data<-CommunityAMARETTO::scHBV_data
#CCLE_Liver_data<-CommunityAMARETTO::CCLE_Liver_data
```

We here create *AMARETTOresults_all*, a named list object with its values as
*AMARETTO_Results* objects corresponding to each dataset. Please note that the
dataset names must be specified and be unique. 

```{r AMARETTO_Run2, message= FALSE,warning=FALSE,results='hide'}

AMARETTOresults_all<-list(TCGA_LIHC=TCGA_LIHC_data$AMARETTO_Results,
                        scHCV=scHCV_data$AMARETTO_Results,
                        scHBV=scHBV_data$AMARETTO_Results)

```


# Running Community-AMARETTO

Community-AMARETTO algorithm consists of three major steps
(and thus three main functions) as follows :

Step 1) *cAMARETTO_Results* function :
This function perform hypergeometric test between every module pairs
across all the input regulatory networks to eveluate the similarity
(or distance) between every module pairs. For each module pair,
the hypergeomteric test outputs p-value, adjusted p-value and
number of overlapping genes. 

```{r AMARETTO_Run3, message= FALSE,warning=FALSE,results='hide'}

cAMARETTOresults<-CommunityAMARETTO::cAMARETTO_Results(AMARETTOresults_all)

```

Step 2) *cAMARETTO_ModuleNetwork* function :
This function construct a network with its nodes representing modules.
We define an edge between each module pair if both of the following
conditions are met: (i) The adjusted p-value computed in Step 1 is
smaller than *pvalue* (default = 0.10) (ii) Number of overlapping genes
between two modules is larger than *inter* (defalut = 5 genes). This defines
a module network where each edge is scored based on the negative 
log-transformed adjusted p-value.

```{r AMARETTO_Run4, message= FALSE,warning=FALSE,results='hide'}

cAMARETTOnetworkM<-CommunityAMARETTO::cAMARETTO_ModuleNetwork(cAMARETTOresults,
                                                    pvalue = 0.10,
                                                    inter = 5,
                                                    plot_network = FALSE)

```

Step 3) cAMARETTOnetworkC function :
We cluster the weighted module network to identify significantly connected
subnetworks using the Girvan-Newman algorithm (Newman and Girvan, 2004).
This algorithm is a divisive method, which aims at detecting subnetworks
by progressively removing edges from the module network according to a score.
The original proposed score is based on the betweenness of edges,
where betweenness is a measure that favors edges that are between
subnetworks, and thus responsible for connecting many pairs of others.

This function implementats the Girvan-Newman algorithm to detect
communities. If *filterComm = TRUE*,  only communities that satisfy the
following conditions will be preserved: number of modules larger than
the 1% of the total number of modules in the network, number of 
represented datasets larger than 10% of the ccommunity size
(and at least, larger than 2) and finally, ratio between edges
inside/outside the subnetwork larger than 1/2.


```{r AMARETTO_Run5, message= FALSE,warning=FALSE,results='hide'}

cAMARETTOnetworkC<-CommunityAMARETTO::cAMARETTO_IdentifyCom(cAMARETTOnetworkM,
                                                    filterComm = FALSE,
                                                    plot_network = FALSE)

```

*cAMARETTOnetworkC* object provides the key information for the 
assignment of communities to modules across different datasets.

Users may optinaly perform the following downstream analysis :


# Functional Enrichments of Communities (Optional)

After detecting communities, we now perform functional enrichment
on each community. This is computed by performing hypergeometric test
between the union of genes inside the community against gensets of our
interest such as hallmark genes of cancer.
We provided a small sample genesets to perform the test. 

```{r AMARETTO_Run6, message= FALSE,warning=FALSE,results='hide'}
geneset_file_address<-system.file("extdata","sample_genesets.gmt",
                                    package = "CommunityAMARETTO")

try(
cAMARETTO_hgtest_tbl<-
    CommunityAMARETTO::CreateHyperGeoTestAll(cAMARETTOresults,
                                            cAMARETTOnetworkM,
                                            cAMARETTOnetworkC,
                    hyper_geo_reference = geneset_file_address,
                    NrCores = 1)
)
```


# HTML-Report for Community-AMARETTO (optinal)

As the last step, the following code will generate an HTML report
for the entire analyis. The reprot provides relationships between
communities, modules and genes.It also provides interactive 
result tables for the functional enrichments of communities and 
phenotype associations of modules. *hyper_geo_reference* and 
*PhenotypeTablesList* parameters of the function are optional. 

Here we provided phenotype association table for each dataset 
to feed it to the html report. However it is optional and
PhenotypeTablesList can simply be assigned to NULL.

```{r AMARETTO_Run7, message= FALSE,warning=FALSE,results='hide'}

PhenotypeTablesList<-list(
                TCGA_LIHC = TCGA_LIHC_data$Phenotype_Association_table,
                #CCLE_Liver=NULL
                scHCV = scHCV_data$Phenotype_Association_table,
                scHBV = scHBV_data$Phenotype_Association_table)

```

The following code generates the html report for the
Community-AMARETTO analysis.

```{r AMARETTO_Run8, message= FALSE,warning=FALSE,results='hide'}
# cAMARETTO_HTMLreport(cAMARETTOresults,
#                     cAMARETTOnetworkM,
#                     cAMARETTOnetworkC,
#                     hyper_geo_reference = cAMARETTO_hgtest_tbl,
#                     output_address = "./",
#                     PhenotypeTablesList = PhenotypeTablesList)
```


# References

1. AMARETTO package, availble in Biocondutor.
2. Ally, Adrian, et al. "Comprehensive and integrative genomic
characterization of hepatocellular carcinoma."
Cell 169.7 (2017): 1327-1341.
3.    Gevaert, O., Villalobos, V., Sikic, B. I. & Plevritis,
S. K. Identification of ovarian cancer driver genes by
using module network integration of multi-omics data.
Interface Focus 3, 20130013–20130013 (2013).



# Session info {.unnumbered}

```{r sessionInfo, echo=FALSE}
sessionInfo()
```