---
title: "Introduction to _CommunityAMARETTO_"
author:
  - name: Mohsen Nabian
    affiliation: Brigham and Women's Hospital, Harvard Medical School, Broad Institute of MIT and Harvard, Boston and Cambridge, MA, USA
  - name: Celine Everaert
    affiliation: Brigham and Women's Hospital, Harvard Medical School, Broad Institute of MIT and Harvard, Boston and Cambridge, MA, USA
  - name: Artur Manukiyan
    affiliation: Brigham and Women's Hospital, Harvard Medical School, Broad Institute of MIT and Harvard, Boston and Cambridge, MA, USA
  - name: Shaimaa Bakr
    affiliation: Stanford Center for Biomedical Informatics Research (BMIR), Department of Medicine and Biomedical Data Science, 1265 Welch Rd, Stanford, CA, USA
  - name: Jishu Xu
    affiliation: Brigham and Women's Hospital, Harvard Medical School, Broad Institute of MIT and Harvard, Boston and Cambridge, MA, USA
  - name: Vincent Carey
    affiliation: Brigham and Women's Hospital, Harvard Medical School, Broad Institute of MIT and Harvard, Boston and Cambridge, MA, USA
    footnote: Corresponding Author
  - name: Olivier Gevaert
    affiliation: Stanford Center for Biomedical Informatics Research (BMIR), Department of Medicine and Biomedical Data Science, 1265 Welch Rd, Stanford, CA, USA
    email: olivier.gevaert@stanford.edu
    footnote: Corresponding Author
  - name: Nathalie Pochet
    affiliation: Brigham and Women's Hospital, Harvard Medical School, Broad Institute of MIT and Harvard, Boston and Cambridge, MA, USA
    email: npochet@broadinstitute.org

date: "`r BiocStyle::doc_date()`"
output:
    BiocStyle::html_document

abstract: |
    The goal of the CommunityAMARETTO algorithm is to identify cell circuits
    and their drivers that are shared and distinct across biological systems.
    Specifically, CommunityAMARETTO takes as input multiple regulatory
    networks inferred using the AMARETTO algorithm [1] that are based
    on multi-omics and clinical/imaging data fusion. Next, CommunityAMARETTO
    learns communities or subnetworks, in particular, regulatory modules
    comprising of cell circuits and their drivers, that are shared and
    distinct across multiple regulatory networks derived from 
    multiple cohorts, diseases, or biological systems more generally,
    using the Girvan-Newman "edge betweenness community detection"
    algorithm (Girvan and Newman, Physical Review E. 2004).

package: Report issues on [https://github.com/broadinstitute/CommunityAMARETTO/] (https://github.com/broadinstitute/CommunityAMARETTO/)
vignette: |
    %\VignetteIndexEntry{Introduction to _CommunityAMARETTO_}
    %\VignetteEngine{knitr::rmarkdown}
    %\VignetteEncoding{UTF-8}
    \usepackage[utf8]{BiocStyle}
---

# Introduction :

Uncovering gene regulatory mechanisms underlying diseases and cancer
systems have long time been sought by many researchers. 
This led to the development of many novel computational algorithms for
regulatory network inference using multi-omics data such as genetics,
epi-genetics and transcriptomics.

These networks attempt to infer system-specific gene regulatory circuits
and thus lead to an enhanced understanding of regulatory
mechanisms as well as discovering biomarkers of diseases and cancer systems.

It is also believed that the integration of multiple system-specific
regulatory networks allows for further discovery
of shared and distinct mechanisms across biological systems. 
To this end, we developed Community-AMARETTO algorithm to integrate multiple
regulatory networks inferred by the AMARETTO algorithm[].
Specifically, Community-AMARETTO algorithm consists of constructing a master network
composed of multiple regulatory networks followed by detecting groups (communities) of circuits that
are shared across systems as well as highliting circuits that are distinct
and system-specific. 

As general guidelines, we here provided instruction for running Community-AMARETTO
on a sample case study of liver cancer.
While here we studied liver-specific data, any cohort/dataset analyzed by AMARETTO
can be used for the Community-AMARETTO analysis. 

Our sample case study includes three biological systems of 1) Liver Cancer,
2) Hepatite C infected liver cancer and 3) Hepatite B infected liver cancer, each
inferred by the AMARETTO algorithm for gene regulatory network inference.

In this note, we show how to integrate these networks through Community-AMARETTO algorithm
, perform downstream analysis such as functional enrichment of detected communities
and to generate interactive HTML-report from the analysis results.


# Installing Package

The following code script shows how to install the package. The package is
accessible via Bioconductor; Alternatively, to access
the newest features of the pacckage, you may install development
branch of github as shown bellow. 


```{r AMARETTO_Run, message= FALSE,warning=FALSE,results='hide'}
library(devtools)
#install.packages("/Users/mohsennabian/Documents/GitHub/CommunityAMARETTO",
#repos = NULL, type = "source")
install_github("broadinstitute/CommunityAMARETTO",
                ref='develop',
                force = 'TRUE')
library(CommunityAMARETTO)
```



# Loading data

CommunityAMARETTO is a network approach developed to integrate multiple
regulatory networks specifically inferred by AMARETTO algorithm. Thus, the required input data
for CommunityAMARETTO algorithm are AMARETTO-result objects (at least two); 
where each AMARETTO-result object is the output of an AMARETTO run on
multi-omics data and contains the information for the inferred 
regulatory network parameters. 

Here as a sample case study, we provided AMARETTO-result objects (outputs for running AMARETTO)
for 3 liver-related datasets : TCGA-LIHC computes resulst from LIHC cohort
of TCGA. scHCV and scHBV are two other single cell studies of Hepatiet C
and B respectively. These sample AMARETTO-result objects can be accessed through the CommunityAMARETTO package as shown below, and are derived from a small portion of the real data, therefore are for representation only. 


```{r AMARETTO_Run1, message= FALSE,warning=FALSE,results='hide'}
#Load data
TCGA_LIHC_data<-CommunityAMARETTO::TCGA_LIHC_data
scHCV_data<-CommunityAMARETTO::scHCV_data
scHBV_data<-CommunityAMARETTO::scHBV_data
#CCLE_Liver_data<-CommunityAMARETTO::CCLE_Liver_data
```

We create AMARETTOresults_all object, which is a list containing our AMARETTO_Results objects,
in this case our 3 regulatory networks.
Please note that the dataset names must be specified and unique. 

```{r AMARETTO_Run2, message= FALSE,warning=FALSE,results='hide'}
AMARETTOresults_all<-list(TCGA_LIHC=TCGA_LIHC_data$AMARETTO_Results,
                        scHCV=scHCV_data$AMARETTO_Results,
                        scHBV=scHBV_data$AMARETTO_Results)
```


# Running Community-AMARETTO

Community-AMARETTO algorithm consists of 
three major steps (and thus three main functions) as follows :

Step 1) cAMARETTO_Results function : This function perform hypergeometric test between every module pairs across all the regulatory networks to eveluate the similarity (or distance) between all modules. For each module pair, the hypergeomteric test outputs p-value, adjusted p-value and number of overlapping genes between each module pair. 

```{r AMARETTO_Run3, message= FALSE,warning=FALSE,results='hide'}

cAMARETTOresults<-CommunityAMARETTO::cAMARETTO_Results(AMARETTOresults_all,
                                                    gmt_filelist = NULL,
                                                    NrCores = 4,
                                                    output_dir = ".",
                                                    drivers = TRUE)
```


Step 2) cAMARETTO_ModuleNetwork function :
This function construct a network of modules (as nodes). We define an edge between each module pair if both of the following conditions are satisfied:
(i) the adjusted p-value computed in Step 1 is smaller than pvalue (default = 0.10) 
(ii) the overlap between two modules is larger than inter (defalut = 5 genes).
This defines a module network where each edge is scored based on the
negative log-transformed adjusted p-value.

```{r AMARETTO_Run4, message= FALSE,warning=FALSE,results='hide'}

cAMARETTOnetworkM<-CommunityAMARETTO::cAMARETTO_ModuleNetwork(cAMARETTOresults,
                                                    pvalue = 0.10,
                                                    inter = 5,
                                                    plot_network = FALSE)
```


Step 3) cAMARETTOnetworkC function :
We cluster the weighted module network to identify significantly connected
subnetworks using the Girvan-Newman algorithm (Newman and Girvan, 2004).
This algorithm is a divisive method, which aims at detecting subnetworks
by progressively removing edges from the module network according to a score.
The original proposed score is based on the betweenness of edges,
where betweenness is a measure that favors edges that are between
subnetworks, and thus responsible for connecting many pairs of others.

This function is implementation of the Girvan-Newman algorithm to detect
communities. Only communities that satisfy the following conditions will be
maintained: number of modules larger than the 1% of the total number 
of modules in the network, number of represented dataset larger than the 10% 
of the ccommunity size (and at least, larger than 2) and finally,
ratio between edges inside/outside the subnetwork larger than 1/2.


```{r AMARETTO_Run5, message= FALSE,warning=FALSE,results='hide'}
cAMARETTOnetworkC<-CommunityAMARETTO::cAMARETTO_IdentifyCom(cAMARETTOnetworkM,
                                                    filterComm = FALSE,
                                                    plot_network = FALSE)
```

cAMARETTOnetworkC object provides the information for the assignment of communities to modules.


# Functional Enrichments of Communities (Optional)

After detecting communities, we now perform functional enrichment
on each community. This is computed by performing hypergeometric test
between the union of genes inside the community and gensets of our interest
such as hallmark genes of cancer.
We provided a small sample genesets to perform the test. 

```{r AMARETTO_Run6, message= FALSE,warning=FALSE,results='hide'}
geneset_file_address<-system.file("extdata","sample_genesets.gmt",
                                    package = "CommunityAMARETTO")

cAMARETTO_hgtest_tbl<-
    CommunityAMARETTO::CreateHyperGeoTestAll(cAMARETTOresults,
                                            cAMARETTOnetworkM,
                                            cAMARETTOnetworkC,
                    hyper_geo_reference = geneset_file_address,
                    NrCores = 1)
```


# HTML-Report for Community-AMARETTO (optinal)

As the last step, the following code will generate an HTML report
for the entire analyis. The reprot provides relationships between
communities, modules and genes. If corresponding tables are given, it provides interactive 
result tables for the functional enrichments of communities and 
phenotype associations of modules. "hyper_geo_reference" and 
"PhenotypeTablesList" parameters of the function are optional. 

Here we provided phenotype association table for each dataset to feed it to the html report. 
However it is potional and PhenotypeTablesList can simply be assigned to NULL.

```{r AMARETTO_Run7, message= FALSE,warning=FALSE,results='hide'}
PhenotypeTablesList<-list(
                TCGA_LIHC = TCGA_LIHC_data$Phenotype_Association_table,
                #CCLE_Liver=NULL
                scHCV = scHCV_data$Phenotype_Association_table,
                scHBV = scHBV_data$Phenotype_Association_table)
```

Here we generate the html report for the entire Community-AMARETTO analysis.

```{r AMARETTO_Run8, message= FALSE,warning=FALSE,results='hide'}
cAMARETTO_HTMLreport(cAMARETTOresults,
                    cAMARETTOnetworkM,
                    cAMARETTOnetworkC,
                    hyper_geo_reference = cAMARETTO_hgtest_tbl,
                    output_address = "./",
                    PhenotypeTablesList = PhenotypeTablesList)
```


# References

1.    Champion, M. et al. Module Analysis Captures Pancancer 
Genetically and Epigenetically Deregulated Cancer Driver Genes
for Smoking and Antiviral Response. EBioMedicine 27, 156–166 (2018).
2.    Gevaert, O., Villalobos, V., Sikic, B. I. & Plevritis,
S. K. Identification of ovarian cancer driver genes by
using module network integration of multi-omics data.
Interface Focus 3, 20130013–20130013 (2013).
3.    Gevaert, O. MethylMix: an R package for identifying
DNA methylation-driven genes. Bioinformatics 31, 1839–1841 (2015).


# Session info {.unnumbered}

```{r sessionInfo, echo=FALSE}
sessionInfo()
```