---
title: "Introduction to _CommunityAMARETTO_"
author:
  - name: Mohsen Nabian
    affiliation: Brigham and Women's Hospital, Harvard Medical School, Broad Institute of MIT and Harvard, Boston and Cambridge, MA, USA
  - name: Celine Everaert
    affiliation: Brigham and Women's Hospital, Harvard Medical School, Broad Institute of MIT and Harvard, Boston and Cambridge, MA, USA
  - name: Artur Manukiyan
    affiliation: Brigham and Women's Hospital, Harvard Medical School, Broad Institute of MIT and Harvard, Boston and Cambridge, MA, USA
  - name: Shaimaa Bakr
    affiliation: Stanford Center for Biomedical Informatics Research (BMIR), Department of Medicine and Biomedical Data Science, 1265 Welch Rd, Stanford, CA, USA
  - name: Jishu Xu
    affiliation: Brigham and Women's Hospital, Harvard Medical School, Broad Institute of MIT and Harvard, Boston and Cambridge, MA, USA
  - name: Vincent Carey
    affiliation: Brigham and Women's Hospital, Harvard Medical School, Broad Institute of MIT and Harvard, Boston and Cambridge, MA, USA
    footnote: Corresponding Author
  - name: Olivier Gevaert
    affiliation: Stanford Center for Biomedical Informatics Research (BMIR), Department of Medicine and Biomedical Data Science, 1265 Welch Rd, Stanford, CA, USA
    email: olivier.gevaert@stanford.edu
    footnote: Corresponding Author
  - name: Nathalie Pochet
    affiliation: Brigham and Women's Hospital, Harvard Medical School, Broad Institute of MIT and Harvard, Boston and Cambridge, MA, USA
    email: npochet@broadinstitute.org


date: "`r BiocStyle::doc_date()`"
output:
  BiocStyle::html_document

abstract: |
     Integrating an increasing number of available multi-omics cancer data remains one of the main challenges to improve our understanding of cancer. One of the main challenges is using multi-omics data for identifying novel cancer driver genes. We have developed an algorithm, called AMARETTO, that integrates copy number, DNA methylation and gene expression data to identify a set of driver genes by analyzing cancer samples and connects them to clusters of co-expressed genes, which we define as modules. We applied AMARETTO in a pancancer setting to identify cancer driver genes and their modules on multiple cancer sites. AMARETTO captures modules enriched in angiogenesis, cell cycle and EMT, and modules that accurately predict survival and molecular subtypes. This allows AMARETTO to identify novel cancer driver genes directing canonical cancer pathways.

package: Report issues on [https://github.com/broadinstitute/CommunityAMARETTO/](https://github.com/broadinstitute/CommunityAMARETTO/)
vignette: |
  %\VignetteIndexEntry{Introduction to _CommunityAMARETTO_}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
  \usepackage[utf8]{BiocStyle}

---

# Installing Package

The package  is accessible via Bioconductor as the most stable version. However, you can install the newest features from the development branch of our github. The package installation should be only done once in your system. Here is the installation code : 

```{r AMARETTO_Run, message= FALSE,warning=FALSE,results='hide'}
library(devtools)
#install.packages("/Users/mohsennabian/Documents/GitHub/CommunityAMARETTO",repos = NULL, type = "source")
devtools::install_github("broadinstitute/CommunityAMARETTO",ref='develop',force = 'TRUE')
library(CommunityAMARETTO)
```

# Loading data

CommunityAMARETTO is a network approach that integrates multiple individual regulatory networks inferred by AMARETTO algorithm. Therefore, the main inputs for CommunityAMARETTO algorithm are multiple AMARETTO-result objects (at least two) corresponding to individual AMARETTO (package) runs. AMARETTO-result is the output of AMARETTO package and it contains the information for the inferred regulatory network.

Here we provided AMARETTO-result objects from 3 liver-related datasets : TCGA-LIHC computes resulst from LIHC cohort of TCGA. scHCV and scHBV are two other single cell studies of Hepatiet C and B respectively. These AMARETTO-result objects are derived from a small portion of the real data and are for representation only. 


```{r AMARETTO_Run2, message= FALSE,warning=FALSE,results='hide'}
#Load data
TCGA_LIHC_data<-CommunityAMARETTO::TCGA_LIHC_data
scHCV_data<-CommunityAMARETTO::scHCV_data
scHBV_data<-CommunityAMARETTO::scHBV_data
#CCLE_Liver_data<-CommunityAMARETTO::CCLE_Liver_data
AMARETTOresults_all<-list(TCGA_LIHC=TCGA_LIHC_data$AMARETTO_Results,
                          scHCV=scHCV_data$AMARETTO_Results,
                          scHBV=scHBV_data$AMARETTO_Results)
```



# Running Community AMARETTO

Here, we present an extension of AMARETTO to a pancancer application using multi-omics data of eleven cancer sites from TCGA. We show that AMARETTO captures modules enriched in major pathways of cancers and modules that accurately predict molecular subtypes. Next, we connect the modules of co-expressed genes in a pancancer module network. We show that this allows the identification of major oncogenic pathways and cancer driver genes involved in multiple cancers. More specifically, we identified a pancancer driver gene that is involved in smoking induced cancers and a pancancer driver gene that is involved in antiviral IFN modulated immune response. Overall, our results show the potential of pancancer multi-omics data fusion to identify cancer drivers that are high within the causal hierarchy of cancer development and associated with common pathways across different types of tumors that eventually can lead to the identification of pancancer drug targets. The AMARETTO algorithm and its pancancer application are publicly available.


2.3 Pancancer Module Network
After running AMARETTO on each cancer site individually, the modules are connected in a pancancer network. Specifically, we evaluate whether there is a significant association between all pairs of modules through a hyper-geometric test. We correct for multiple hypothesis testing using the false discovery rate (Benjamini and Hochberg, 1995). We consider the association to be significant if both of the following conditions are satisfied: (i) the adjusted p-value is smaller than 0.05 and (ii) the overlap between two modules is larger than 5 genes. This defines a module network where each edge is scored based on the negative log-transformed adjusted p-value.



The communityAMARETTO algorithm comprises of three major functions :

1) cAMARETTO_Results : This function performs hypergeormtric test for all existing module pairs across all datasets. P-value extracted for each hypergeoetric test quantifies the signicance of gene overlaping between each module pair.

2) cAMARETTO_ModuleNetwork : This function forms a network of connected modules with edges representing significant overlapping genes between connected modules. The thresholds for existance or non-existance of an edge between each module pair can be set by : a) "pvalue" computed between each two modules in previous step and b) "inter" which is the minimum number of overlapping genes requiered to consider an edge between the module pairs. 

3) cAMARETTOnetworkC : 
We cluster the weighted module network to identify significantly connected subnetworks using the Girvan-Newman algorithm (Newman and Girvan, 2004). This algorithm is a divisive method, which aims at detecting subnetworks by progressively removing edges from the module network according to a score. The original proposed score is based on the betweenness of edges, where betweenness is a measure that favors edges that are between subnetworks, and thus responsible for connecting many pairs of others.

This function is implementation of the Girvan-Newman algorithm to detect communities. Only communities that satisfy the following conditions will be maintained: number of modules larger than the 1% of the total number of modules in the network, number of represented dataset larger than the 10% of the ccommunity size (and at least, larger than 2) and finally, ratio between edges inside/outside the subnetwork larger than 1/2.


```{r AMARETTO_Run3, message= FALSE,warning=FALSE,results='hide'}

cAMARETTOresults<-CommunityAMARETTO::cAMARETTO_Results(AMARETTOresults_all,
                                                       gmt_filelist=NULL,
                                                       NrCores = 4,
                                                       output_dir = ".",
                                                       drivers=TRUE)

cAMARETTOnetworkM<-cAMARETTO_ModuleNetwork(cAMARETTOresults,
                                           pvalue=0.10,
                                           inter = 5)

cAMARETTOnetworkC<-cAMARETTO_IdentifyCom(cAMARETTOnetworkM,filterComm = FALSE)

```

# Functional Enrichments of Communities

After detecting communities, we now perform functional enrichment on each community. This is computed by performing hypergeometric test between the union of genes inside the community and gensets of our interests such as hallmark genes of cancer. We provided a small sample genesets to perform the test. 

```{r AMARETTO_Run4, message= FALSE,warning=FALSE,results='hide'}
geneset_file_address<-system.file("extdata",
                                  "sample_genesets.gmt",
                                  package = "CommunityAMARETTO")

cAMARETTO_hgtest_tbl<-CommunityAMARETTO::CreateHyperGeoTestAll(cAMARETTOresults,
                                                               cAMARETTOnetworkM,
                                                               cAMARETTOnetworkC,
                                        hyper_geo_reference=geneset_file_address,
                                                               NrCores=1)

```


# HTML-Report for Community-AMARETTO

As the last step, the following code will generate an HTML report for the entire analyis. The reprot project relationships between communities, modules and genes as well as it provides interactive result tables for the functional enrichments of communities and phenotype associations of modules.  "hyper_geo_reference" and "PhenotypeTablesList" parameters of the function are optional. 

```{r AMARETTO_Run6, message= FALSE,warning=FALSE,results='hide'}
cAMARETTO_HTMLreport(cAMARETTOresults,
                     cAMARETTOnetworkM,
                     cAMARETTOnetworkC,
                     hyper_geo_reference = cAMARETTO_hgtest_tbl,
                     output_address= "./",
                     PhenotypeTablesList=PhenotypeTablesList)
```


# References

1.	Champion, M. et al. Module Analysis Captures Pancancer Genetically and Epigenetically Deregulated Cancer Driver Genes for Smoking and Antiviral Response. EBioMedicine 27, 156–166 (2018).
2.	Gevaert, O., Villalobos, V., Sikic, B. I. & Plevritis, S. K. Identification of ovarian cancer driver genes by using module network integration of multi-omics data. Interface Focus 3, 20130013–20130013 (2013).
3.	Gevaert, O. MethylMix: an R package for identifying DNA methylation-driven genes. Bioinformatics 31, 1839–1841 (2015).

# Appendix 

\begin{center}
\begin{tabular}{|l|l|l|l|}
\hline
Cancer code & Cancer site  \\
\hline
\\
BLCA & bladder urothelial carcinoma  \\
BRCA & breast invasive carcinoma  \\
CESC & cervical squamous cell carcinoma and endocervical adenocarcinoma\\
CHOL & cholangiocarcinoma\\
COAD & colon adenocarcinoma \\
ESCA & esophageal carcinoma\\
GBM & glioblastoma multiforme\\
HNSC & head and neck squamous cell carcinoma \\
KIRC & kidney renal clear cell carcinoma  \\
KIRP & kidney renal papillary cell carcinoma\\
LAML & acute myeloid leukemia \\
LGG & brain lower grade glioma\\
LIHC & liver hepatocellular carcinoma\\
LUAD & lung adenocarcinoma \\
LUSC & lung squamous cell carcinoma \\
OV & ovarian serous cystadenocarcinoma \\
PAAD & pancreatic adenocarcinoma\\
PCPG & pheochromocytoma and paraganglioma\\
READ & rectum adenocarcinoma \\
SARC & sarcoma\\
STAD & stomach adenocarcinoma\\
THCA & thyroid carcinoma\\
THYM & thymoma\\
UCEC & endometrial carcinoma \\
\\
%\hdashline
COADREAD & colon cancer + rectal cancer \\
\hline
\end{tabular}
\end{center}



\newpage

# Session info {.unnumbered}

```{r sessionInfo, echo=FALSE}
sessionInfo()
```