\documentclass[8pt,a4,]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
  \newcommand{\euro}{â‚¬}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\usepackage[margin = 1in]{geometry}


\usepackage{longtable,booktabs}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

\RequirePackage[]{/Library/Frameworks/R.framework/Versions/3.5/Resources/library/BiocStyle/resources/tex/Bioconductor}

% Create subtitle command for use in maketitle
\newcommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}

\bioctitle[]{Introduction to \emph{community-AMARETTO}}
    \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
\author[1]{Jayendra Shinde}
\author[2]{Celine Everaert}
\author[1]{Shaimaa Bakr}
\author[2]{Mohsen Nabian}
\author[2]{Jishu Xu}
\author[2]{Nathalie Pochet\thanks{\ttfamily npochet@broadinstitute.org}}
\author[1]{Olivier Gevaert\thanks{\ttfamily olivier.gevaert@stanford.edu}}
\affil[1]{Stanford Center for Biomedical Informatics Research (BMIR), Department of Medicine and Biomedical Data Science, 1265 Welch Rd, Stanford, CA, USA}
\affil[2]{Brigham and Women's Hospital, Harvard Medical School, Broad Institute of MIT and Harvard, Cambridge, MA, USA}
    \preauthor{\centering\large\emph}
  \postauthor{\par}
      \predate{\centering\large\emph}
  \postdate{\par}
    \date{22 March 2019}


% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

% code highlighting
\definecolor{fgcolor}{rgb}{0.251, 0.251, 0.251}
\newcommand{\hlnum}[1]{\textcolor[rgb]{0.816,0.125,0.439}{#1}}%
\newcommand{\hlstr}[1]{\textcolor[rgb]{0.251,0.627,0.251}{#1}}%
\newcommand{\hlcom}[1]{\textcolor[rgb]{0.502,0.502,0.502}{\textit{#1}}}%
\newcommand{\hlopt}[1]{\textcolor[rgb]{0,0,0}{#1}}%
\newcommand{\hlstd}[1]{\textcolor[rgb]{0.251,0.251,0.251}{#1}}%
\newcommand{\hlkwa}[1]{\textcolor[rgb]{0.125,0.125,0.941}{#1}}%
\newcommand{\hlkwb}[1]{\textcolor[rgb]{0,0,0}{#1}}%
\newcommand{\hlkwc}[1]{\textcolor[rgb]{0.251,0.251,0.251}{#1}}%
\newcommand{\hlkwd}[1]{\textcolor[rgb]{0.878,0.439,0.125}{#1}}%
\let\hlipl\hlkwb
%
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
%
\newenvironment{Shaded}{\begin{myshaded}}{\end{myshaded}}
% set background for result chunks
\let\oldverbatim\verbatim
\renewenvironment{verbatim}{\color{codecolor}\begin{myshaded}\begin{oldverbatim}}{\end{oldverbatim}\end{myshaded}}
%
\newcommand{\KeywordTok}[1]{\hlkwd{#1}}
\newcommand{\DataTypeTok}[1]{\hlkwc{#1}}
\newcommand{\DecValTok}[1]{\hlnum{#1}}
\newcommand{\BaseNTok}[1]{\hlnum{#1}}
\newcommand{\FloatTok}[1]{\hlnum{#1}}
\newcommand{\ConstantTok}[1]{\hlnum{#1}}
\newcommand{\CharTok}[1]{\hlstr{#1}}
\newcommand{\SpecialCharTok}[1]{\hlstr{#1}}
\newcommand{\StringTok}[1]{\hlstr{#1}}
\newcommand{\VerbatimStringTok}[1]{\hlstr{#1}}
\newcommand{\SpecialStringTok}[1]{\hlstr{#1}}
\newcommand{\ImportTok}[1]{{#1}}
\newcommand{\CommentTok}[1]{\hlcom{#1}}
\newcommand{\DocumentationTok}[1]{\hlcom{#1}}
\newcommand{\AnnotationTok}[1]{\hlcom{#1}}
\newcommand{\CommentVarTok}[1]{\hlcom{#1}}
\newcommand{\OtherTok}[1]{{#1}}
\newcommand{\FunctionTok}[1]{\hlstd{#1}}
\newcommand{\VariableTok}[1]{\hlstd{#1}}
\newcommand{\ControlFlowTok}[1]{\hlkwd{#1}}
\newcommand{\OperatorTok}[1]{\hlopt{#1}}
\newcommand{\BuiltInTok}[1]{{#1}}
\newcommand{\ExtensionTok}[1]{{#1}}
\newcommand{\PreprocessorTok}[1]{\textit{#1}}
\newcommand{\AttributeTok}[1]{{#1}}
\newcommand{\RegionMarkerTok}[1]{{#1}}
\newcommand{\InformationTok}[1]{\textcolor{messagecolor}{#1}}
\newcommand{\WarningTok}[1]{\textcolor{warningcolor}{#1}}
\newcommand{\AlertTok}[1]{\textcolor{errorcolor}{#1}}
\newcommand{\ErrorTok}[1]{\textcolor{errorcolor}{#1}}
\newcommand{\NormalTok}[1]{\hlstd{#1}}
%
\AtBeginDocument{\bibliographystyle{/Library/Frameworks/R.framework/Versions/3.5/Resources/library/BiocStyle/resources/tex/unsrturl}}
\usepackage{booktabs}
\usepackage{longtable}
\usepackage{array}
\usepackage{multirow}
\usepackage{wrapfig}
\usepackage{float}
\usepackage{colortbl}
\usepackage{pdflscape}
\usepackage{tabu}
\usepackage{threeparttable}
\usepackage{threeparttablex}
\usepackage[normalem]{ulem}
\usepackage{makecell}
\usepackage{xcolor}

\begin{document}
\maketitle
\begin{abstract}
The goal of the Community-AMARETTO algorithm (Champion et al.,
EBioMedicine 2018) is to identify cell circuits and their drivers that
are shared and distinct across biological systems. Specifically,
Community-AMARETTO takes as input multiple regulatory networks inferred
using the AMARETTO algorithm that are based on multi-omics and imaging
data fusion. Next, Community-AMARETTO learns communities or subnetworks,
in particular, regulatory modules comprising of cell circuits and their
drivers, that are shared and distinct across multiple regulatory
networks derived from multiple cohorts, diseases, or biological systems
more generally, using the Girvan-Newman ``edge betweenness community
detection'' algorithm (Girvan and Newman, Physical Review E. 2004).
\end{abstract}

\packageVersion{Report issues on
\url{https://github.com/broadinstitute/CommunityAMARETTO}}

{
\setcounter{tocdepth}{2}
\newpage
\tableofcontents
\newpage
}
\section{Introduction}\label{introduction}

The package \emph{AMARETTO} contains functions to use the statistical
algorithm AMARETTO, an algorithm to identify cancer drivers by
integrating a variety of omics data from cancer and normal tissue. Due
to the increasing availability of multi-omics data sets, there is a need
for computational methods that integrate multi-omics data set and create
knowl-edge. Especially in the field of cancer research, large
international projects such as The Cancer Genome Atlas (TCGA) and the
International Cancer Genome Consortium (ICGC) are producing large
quantities of multi-omics data for each cancer site. However it remains
unknown which profile is the most meaningful and how to effciently
integrate different omics profiles. AMARETTO is an algorithm to unravel
cancer drivers by reducing the data dimensionality into cancer modules.
AMARETTO first models the effects of genomic/epigenomic data on disease
specific gene expression. AMARETTO's second step involves constructing
co-expressed modules to connect the cancer drivers with their downstream
targets. We applied AMARETTO to several cancer sites of the TCGA project
allowing to identify several cancer driver genes of interest, including
novel genes in addition to known drivers of cancer. This package also
includes functionality to access TCGA data directly so the user can
immediately run AMARETTO on the most recent version of the data.

\section{Installation Instructions}\label{installation-instructions}

To install AMARETTO is to first download the appropriate file for your
platform from the Bioconductor website
\url{http://www.bioconductor.org/}. For Windows, start R and select the
Packages menu, then Install package from local zip file. Find and
highlight the location of the zip file and click on open. For
Linux/Unix, use the usual command R CMD INSTALL or install from
bioconductor

The package can be installed from the GitHub repository :

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#-------------------------------------------------------------------------------}

\KeywordTok{install.packages}\NormalTok{(}\StringTok{"BiocManager"}\NormalTok{,}\DataTypeTok{repos =} \StringTok{"http://cran.us.r-project.org"}\NormalTok{)}
\NormalTok{BiocManager}\OperatorTok{::}\KeywordTok{install}\NormalTok{(}\StringTok{"gevaertlab/AMARETTO"}\NormalTok{)}

\CommentTok{#-------------------------------------------------------------------------------}
\end{Highlighting}
\end{Shaded}

Help files. Detailed information on AMARETTO package functions can be
obtained in the help files. For example, to view the help file for the
function AMARETTO in an R session, use ?AMARETTO.

\section{Data Input}\label{data-input}

\subsection{Data Access}\label{data-access}

\begin{center}
\begin{tabular}{|l|l|l|l|}
\hline
Cancer code & Cancer site  \\
\hline
\\
BLCA & bladder urothelial carcinoma  \\
BRCA & breast invasive carcinoma  \\
CESC & cervical squamous cell carcinoma and endocervical adenocarcinoma\\
CHOL & cholangiocarcinoma\\
COAD & colon adenocarcinoma \\
ESCA & esophageal carcinoma\\
GBM & glioblastoma multiforme\\
HNSC & head and neck squamous cell carcinoma \\
KIRC & kidney renal clear cell carcinoma  \\
KIRP & kidney renal papillary cell carcinoma\\
LAML & acute myeloid leukemia \\
LGG & brain lower grade glioma\\
LIHC & liver hepatocellular carcinoma\\
LUAD & lung adenocarcinoma \\
LUSC & lung squamous cell carcinoma \\
OV & ovarian serous cystadenocarcinoma \\
PAAD & pancreatic adenocarcinoma\\
PCPG & pheochromocytoma and paraganglioma\\
READ & rectum adenocarcinoma \\
SARC & sarcoma\\
STAD & stomach adenocarcinoma\\
THCA & thyroid carcinoma\\
THYM & thymoma\\
UCEC & endometrial carcinoma \\
\\
%\hdashline
COADREAD & colon cancer + rectal cancer \\
\hline
\end{tabular}
\end{center}

\newpage

We also added COADREAD as a combination of colon and rectal cancer, as
reports have shown that both can be seen as a single disease. The cancer
code is needed to download data from TCGA and one needs to decide on a
target location to save the data locally in the TargetDirectory,
e.g.~the /Downloads/ folder on a mac.

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#-------------------------------------------------------------------------------}
\CommentTok{#Load AMARETTO output files on the two runs }
\KeywordTok{data}\NormalTok{(AMARETTOinit1_tutorial,AMARETTOresults1_tutorial)}
\KeywordTok{data}\NormalTok{(AMARETTOinit2_tutorial,AMARETTOresults2_tutorial)}

\CommentTok{#-------------------------------------------------------------------------------}
\end{Highlighting}
\end{Shaded}

We recommend to use one TargetDirectory for all cancer sites, as this
will save all data in one hierarchy is convenient when revisting results
later on. The directory structure that is created will also include the
data version history, so it is easy to report what version of the data
is used. AMARETTO\_Download() downloads the data, extracts archives and
provides the paths to the downloaded folder for preprocessing.
AMARETTO\_Download() can also be run without actually downloading the
data as follows:

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#-------------------------------------------------------------------------------}



\CommentTok{#-------------------------------------------------------------------------------}
\end{Highlighting}
\end{Shaded}

This is convenient when revisiting a data set because you want to
redo-downstream analysis, but not the actual downloading. Running this
way, will only set the data paths. The next step is preprocessing.

\subsection{DNA Methylation Data}\label{dna-methylation-data}

DNA methylation data has to be run by MethylMix which is also
computationally intensive and therefore we have chosen to provide add
the MethylMix output to the AMARETTO package instead of processing the
raw DNA methylation data. This functionality is available in the
\href{https://www.bioconductor.org/packages/release/bioc/html/MethylMix.html}{MethylMix
package}

\newpage

\subsection{Data Preprocessing}\label{data-preprocessing}

The data preprocessing step will take care of preprocessing the gene
expression and DNA copy number data. Data preprocessing is done by
Preprocess CancerSite which takes the CancerSite and the data set
directories as parameters:

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#-------------------------------------------------------------------------------}


\CommentTok{#-------------------------------------------------------------------------------}
\end{Highlighting}
\end{Shaded}

This function preprocessed the gene expression data and the DNA copy
number data. For the gene expression data, different preprocessing is
done for microarray and RNA sequencing data. This involves missing value
estimation using K-nearest neighbors. Also genes or patients that have
more than 10\% missing values are removed. Next, batch correction is
done using the Combat method. For certain cancer sites, the gene
expression data is split up in separate sub-data sets. This function
first uses the preprocessing pipeline on each sub-data set separately
and combines the data afterwards. For the DNA copy number data, the
GISTIC algorithm output data is used. All genes that are in
amplifications or deletions based on GISTIC output are extracted and the
segmented DNA copy number data is stored. The segmented DNA copy number
data is also batch corrected using Combat.

\begin{verbatim}
## Warning: package 'stringr' was built under R version 3.5.2
## Warning: package 'tibble' was built under R version 3.5.2
## Warning: package 'purrr' was built under R version 3.5.2
## Warning: package 'dplyr' was built under R version 3.5.2
## Warning: replacing previous import 'callr::run' by 'rmarkdown::run' when
## loading 'AMARETTO'
## Warning: replacing previous import 'DT::dataTableOutput' by
## 'shiny::dataTableOutput' when loading 'AMARETTO'
## Warning: replacing previous import 'DT::renderDataTable' by
## 'shiny::renderDataTable' when loading 'AMARETTO'
\end{verbatim}

\section{Running AMARETTO}\label{running-amaretto}

In the case that you run AMARETTO with your own data, three data
matrices are needed with preprocessed gene expression, DNA copy number
and DNA methylation data, where genes are in the rows and patients are
in the columns. Once you have your own data in this format or using a
previously downloaded TCGA data set, you can start doing analysis with
AMARETTO. First, we need to initialize the algorithm by clustering the
gene expression data and creating the regulator data object. This is
done by the AMARETTO Initialize function and the TCGA LIHC data set:

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#-------------------------------------------------------------------------------}



\CommentTok{#-------------------------------------------------------------------------------}
\end{Highlighting}
\end{Shaded}

Besides the three data sets, you need to decide how many modules to
build and how much of the gene expression data is going to be used. For
a first run we recommend 100 modules and to use the top 25\% most
varying genes. The AMARETTOinit object now contains cluster information
to initialize an AMARETTO run and also stores the parameters that are
required for AMARETTO. Now we can run AMARETTO as follows:

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#-------------------------------------------------------------------------------}


\CommentTok{#-------------------------------------------------------------------------------}
\end{Highlighting}
\end{Shaded}

This can take anywhere from 10 minutes up to 1 hour to build the modules
for the TCGA cohorts depending on the number of genes that is modeled
and the number of patients that is available. The breast cancer data set
(BRCA) is the largest data set and will take the longest time to
converge. AMARETTO will stop when less than 1\% of the genes are
reassigned to other modules. Next, one can test the AMARETTO model on
the training set by calculating the Rsquare for each module using the
AMARETTO EvaluateTestSet function:

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#-------------------------------------------------------------------------------}



\CommentTok{#-------------------------------------------------------------------------------}
\end{Highlighting}
\end{Shaded}

This function will use the training data to calculate the performance
for predicting genes expression values based on the selected regulators.
Of course, it is more interesting to use an independent test set. In
this case only a gene expression data set is needed, for example from
the GEO database. This will allow to check how well the AMARETTO modules
are generalizing to new data. Take care that the an independent data set
needs to be centered and scaled to unit variance. The AMARETTOtestReport
will also give information of how many regulators and cluster members
are actually present. The Rsquare performance has to be interpreted in
this context as if many regulators are absent in the test data set due
to platform limitations, the performance will be adversely affected.
Finally, modules can be visualized using the AMARETTO VisualizeModule
function:

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#-------------------------------------------------------------------------------}
\end{Highlighting}
\end{Shaded}

Additionaly, to a standard version of the heatmap, one can add sample
annotations to interogate biological phenotypes.

\newpage

\section{HTML Report of AMARETTO}\label{html-report-of-amaretto}

To retrieve heatmaps for all of the modules and additional tables with
gene set enrichment data one can run a HTML report.

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#-------------------------------------------------------------------------------}



\CommentTok{#-------------------------------------------------------------------------------}
\end{Highlighting}
\end{Shaded}

\section{References}\label{references}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  Champion, M. et al. Module Analysis Captures Pancancer Genetically and
  Epigenetically Deregulated Cancer Driver Genes for Smoking and
  Antiviral Response. EBioMedicine 27, 156--166 (2018).
\item
  Gevaert, O., Villalobos, V., Sikic, B. I. \& Plevritis, S. K.
  Identification of ovarian cancer driver genes by using module network
  integration of multi-omics data. Interface Focus 3, 20130013--20130013
  (2013).
\item
  Gevaert, O. MethylMix: an R package for identifying DNA
  methylation-driven genes. Bioinformatics 31, 1839--1841 (2015).
\end{enumerate}

\newpage

\section*{Session info}\label{session-info}
\addcontentsline{toc}{section}{Session info}

\begin{verbatim}
## R version 3.5.1 (2018-07-02)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS  10.14.2
## 
## Matrix products: default
## BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] grid      parallel  stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] AMARETTO_0.99.1            rmarkdown_1.11            
##  [3] reshape2_1.4.3             htmltools_0.3.6           
##  [5] DT_0.5                     Rcpp_1.0.0                
##  [7] callr_3.1.1.9000           forcats_0.3.0             
##  [9] dplyr_0.8.0.1              purrr_0.3.1               
## [11] readr_1.3.1                tidyr_0.8.2               
## [13] tibble_2.0.1               ggplot2_3.1.0             
## [15] tidyverse_1.2.1            TCGAutils_1.0.1           
## [17] curatedTCGAData_1.2.0      MultiAssayExperiment_1.6.0
## [19] randomcoloR_1.1.0          R.utils_2.7.0             
## [21] R.oo_1.22.0                R.methodsS3_1.7.1         
## [23] circlize_0.4.5             ComplexHeatmap_1.20.0     
## [25] stringr_1.4.0              impute_1.56.0             
## [27] RColorBrewer_1.1-2         matrixStats_0.54.0        
## [29] glmnet_2.0-16              Matrix_1.2-14             
## [31] doParallel_1.0.14          iterators_1.0.10          
## [33] foreach_1.4.4              limma_3.38.3              
## [35] RCurl_1.95-4.11            bitops_1.0-6              
## [37] BiocStyle_2.8.2           
## 
## loaded via a namespace (and not attached):
##   [1] tidyselect_0.2.5              lme4_1.1-19                  
##   [3] RSQLite_2.1.1                 AnnotationDbi_1.44.0         
##   [5] htmlwidgets_1.3               BiocParallel_1.16.2          
##   [7] gmp_0.5-13.2                  Rtsne_0.15                   
##   [9] munsell_0.5.0                 codetools_0.2-15             
##  [11] withr_2.1.2                   colorspace_1.4-0             
##  [13] BiocInstaller_1.30.0          Biobase_2.42.0               
##  [15] knitr_1.21                    rstudioapi_0.9.0             
##  [17] geometry_0.3-6                stats4_3.5.1                 
##  [19] FD_1.0-12                     GenomeInfoDbData_1.2.0       
##  [21] KMsurv_0.1-5                  bit64_0.9-7                  
##  [23] generics_0.0.2                xfun_0.4                     
##  [25] R6_2.4.0                      GenomeInfoDb_1.18.1          
##  [27] rsvd_1.0.0                    DelayedArray_0.8.0           
##  [29] assertthat_0.2.0              promises_1.0.1               
##  [31] scales_1.0.0                  gtable_0.2.0                 
##  [33] sva_3.30.0                    processx_3.2.1               
##  [35] rlang_0.3.1                   genefilter_1.64.0            
##  [37] cmprsk_2.2-7                  GlobalOptions_0.1.0          
##  [39] splines_3.5.1                 lazyeval_0.2.1               
##  [41] broom_0.5.1                   yaml_2.2.0                   
##  [43] abind_1.4-5                   modelr_0.1.3                 
##  [45] backports_1.1.3               httpuv_1.4.5.1               
##  [47] qvalue_2.14.1                 tools_3.5.1                  
##  [49] bookdown_0.9                  kableExtra_1.0.1             
##  [51] BiocGenerics_0.28.0           plyr_1.8.4                   
##  [53] zlibbioc_1.28.0               ps_1.3.0                     
##  [55] ggpubr_0.2                    GetoptLong_0.1.7             
##  [57] zoo_1.8-4                     S4Vectors_0.20.1             
##  [59] SummarizedExperiment_1.12.0   haven_2.0.0                  
##  [61] cluster_2.0.7-1               magrittr_1.5                 
##  [63] data.table_1.12.0             survminer_0.4.3              
##  [65] hms_0.4.2                     mime_0.6                     
##  [67] evaluate_0.13                 xtable_1.8-3                 
##  [69] XML_3.98-1.17                 readxl_1.2.0                 
##  [71] IRanges_2.16.0                gridExtra_2.3                
##  [73] shape_1.4.4                   snm_1.30.0                   
##  [75] compiler_3.5.1                V8_2.0                       
##  [77] crayon_1.3.4                  minqa_1.2.4                  
##  [79] mgcv_1.8-24                   corpcor_1.6.9                
##  [81] later_0.8.0                   lubridate_1.7.4              
##  [83] DBI_1.0.0                     ExperimentHub_1.6.1          
##  [85] ClusterR_1.1.8                magic_1.5-9                  
##  [87] MASS_7.3-50                   rappdirs_0.3.1               
##  [89] ade4_1.7-13                   permute_0.9-4                
##  [91] cli_1.0.1                     GenomicRanges_1.34.0         
##  [93] pkgconfig_2.0.2               km.ci_0.5-2                  
##  [95] xml2_1.2.0                    annotate_1.60.0              
##  [97] webshot_0.5.1                 XVector_0.22.0               
##  [99] rvest_0.3.2                   digest_0.6.18                
## [101] vegan_2.5-3                   cellranger_1.1.0             
## [103] survMisc_0.5.5                curl_3.3                     
## [105] shiny_1.2.0                   gtools_3.8.1                 
## [107] rjson_0.2.20                  nloptr_1.2.1                 
## [109] nlme_3.1-137                  GenomicDataCommons_1.4.3     
## [111] jsonlite_1.6                  edge_2.14.0                  
## [113] viridisLite_0.3.0             pillar_1.3.1                 
## [115] lattice_0.20-35               httr_1.4.0                   
## [117] survival_2.43-3               interactiveDisplayBase_1.18.0
## [119] glue_1.3.0                    shinythemes_1.1.2            
## [121] bit_1.1-14                    stringi_1.3.1                
## [123] blob_1.1.1                    lfa_1.12.0                   
## [125] AnnotationHub_2.12.1          jackstraw_1.3                
## [127] memoise_1.1.0                 irlba_2.3.2                  
## [129] ape_5.2
\end{verbatim}

\end{document}
