---
title: "Introduction to _CommunityAMARETTO_"
author:
  - name: Mohsen Nabian
    affiliation: Brigham and Women's Hospital, Harvard Medical School, Broad Institute of MIT and Harvard, Boston and Cambridge, MA, USA
  - name: Celine Everaert
    affiliation: Brigham and Women's Hospital, Harvard Medical School, Broad Institute of MIT and Harvard, Boston and Cambridge, MA, USA
  - name: Artur Manukiyan
    affiliation: Brigham and Women's Hospital, Harvard Medical School, Broad Institute of MIT and Harvard, Boston and Cambridge, MA, USA
  - name: Shaimaa Bakr
    affiliation: Stanford Center for Biomedical Informatics Research (BMIR), Department of Medicine and Biomedical Data Science, 1265 Welch Rd, Stanford, CA, USA
  - name: Jishu Xu
    affiliation: Brigham and Women's Hospital, Harvard Medical School, Broad Institute of MIT and Harvard, Boston and Cambridge, MA, USA
  - name: Vincent Carey
    affiliation: Brigham and Women's Hospital, Harvard Medical School, Broad Institute of MIT and Harvard, Boston and Cambridge, MA, USA
    footnote: Corresponding Author
  - name: Olivier Gevaert
    affiliation: Stanford Center for Biomedical Informatics Research (BMIR), Department of Medicine and Biomedical Data Science, 1265 Welch Rd, Stanford, CA, USA
    email: olivier.gevaert@stanford.edu
    footnote: Corresponding Author
  - name: Nathalie Pochet
    affiliation: Brigham and Women's Hospital, Harvard Medical School, Broad Institute of MIT and Harvard, Boston and Cambridge, MA, USA
    email: npochet@broadinstitute.org


date: "`r BiocStyle::doc_date()`"
output:
  BiocStyle::html_document

abstract: |
     Integrating an increasing number of available multi-omics cancer data remains one of the main challenges to improve our understanding of cancer. One of the main challenges is using multi-omics data for identifying novel cancer driver genes. We have developed an algorithm, called AMARETTO, that integrates copy number, DNA methylation and gene expression data to identify a set of driver genes by analyzing cancer samples and connects them to clusters of co-expressed genes, which we define as modules. We applied AMARETTO in a pancancer setting to identify cancer driver genes and their modules on multiple cancer sites. AMARETTO captures modules enriched in angiogenesis, cell cycle and EMT, and modules that accurately predict survival and molecular subtypes. This allows AMARETTO to identify novel cancer driver genes directing canonical cancer pathways.

package: Report issues on [https://github.com/broadinstitute/CommunityAMARETTO/](https://github.com/broadinstitute/CommunityAMARETTO/)
vignette: |
  %\VignetteIndexEntry{Introduction to _CommunityAMARETTO_}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
  \usepackage[utf8]{BiocStyle}

---

```{r AMARETTO_Run, message= FALSE,warning=FALSE,results='hide'}
install.packages("/Users/mohsennabian/Documents/GitHub/CommunityAMARETTO",repos = NULL, type = "source")
library(CommunityAMARETTO)
#Load data
TCGA_LIHC_data<-CommunityAMARETTO::TCGA_LIHC_data
scHCV_data<-CommunityAMARETTO::scHCV_data
scHBV_data<-CommunityAMARETTO::scHBV_data
#CCLE_Liver_data<-CommunityAMARETTO::CCLE_Liver_data
AMARETTOresults_all<-list(TCGA_LIHC=TCGA_LIHC_data$AMARETTO_Results,
                          scHCV=scHCV_data$AMARETTO_Results,
                          scHBV=scHBV_data$AMARETTO_Results)

PhenotypeTablesList<-list(TCGA_LIHC=TCGA_LIHC_data$Phenotype_Association_table,
                          scHCV=scHCV_data$Phenotype_Association_table,
                          scHBV=scHBV_data$Phenotype_Association_table)

# AMARETTOresults_all<-list(TCGA_LIHC=TCGA_LIHC_data$AMARETTO_Results,
#                           CCLE_Liver=CCLE_Liver_data$AMARETTO_Results)
# PhenotypeTablesList<-list(TCGA_LIHC=TCGA_LIHC_data$Phenotype_Association_table,
#                           CCLE_Liver=NULL)
cAMARETTOresults<-CommunityAMARETTO::cAMARETTO_Results(AMARETTOresults_all,
                                                       gmt_filelist=NULL,
                                                       NrCores = 4,
                                                       output_dir = ".",
                                                       drivers=TRUE)

cAMARETTOnetworkM<-cAMARETTO_ModuleNetwork(cAMARETTOresults,0.10,5)
cAMARETTOnetworkC<-cAMARETTO_IdentifyCom(cAMARETTOnetworkM,filterComm = FALSE)
geneset_file_address<-system.file("extdata",
                                  "sample_genesets.gmt",
                                  package = "CommunityAMARETTO")
cAMARETTO_hgtest_tbl<-CommunityAMARETTO::CreateHyperGeoTestAll(cAMARETTOresults,
                                                               cAMARETTOnetworkM,
                                                               cAMARETTOnetworkC,
                                        hyper_geo_reference=geneset_file_address,
                                                               driverGSEA=TRUE,
                                                               NrCores=1)
cAMARETTO_HTMLreport(cAMARETTOresults,
                     cAMARETTOnetworkM,
                     cAMARETTOnetworkC,
                     hyper_geo_reference = cAMARETTO_hgtest_tbl,
                     output_address= "./",
                     PhenotypeTablesList=PhenotypeTablesList)
```



# HTML-Report for Community AMARETTO

```{r AMARETTO_Run2, message= FALSE,warning=FALSE,results='hide'}
cAMARETTO_HTMLreport(cAMARETTOresults,
                     cAMARETTOnetworkM,
                     cAMARETTOnetworkC,
                     hyper_geo_reference = cAMARETTO_hgtest_tbl,
                     output_address= "./",
                     PhenotypeTablesList=PhenotypeTablesList)
```



# References

1.	Champion, M. et al. Module Analysis Captures Pancancer Genetically and Epigenetically Deregulated Cancer Driver Genes for Smoking and Antiviral Response. EBioMedicine 27, 156–166 (2018).
2.	Gevaert, O., Villalobos, V., Sikic, B. I. & Plevritis, S. K. Identification of ovarian cancer driver genes by using module network integration of multi-omics data. Interface Focus 3, 20130013–20130013 (2013).
3.	Gevaert, O. MethylMix: an R package for identifying DNA methylation-driven genes. Bioinformatics 31, 1839–1841 (2015).

# Appendix 

\begin{center}
\begin{tabular}{|l|l|l|l|}
\hline
Cancer code & Cancer site  \\
\hline
\\
BLCA & bladder urothelial carcinoma  \\
BRCA & breast invasive carcinoma  \\
CESC & cervical squamous cell carcinoma and endocervical adenocarcinoma\\
CHOL & cholangiocarcinoma\\
COAD & colon adenocarcinoma \\
ESCA & esophageal carcinoma\\
GBM & glioblastoma multiforme\\
HNSC & head and neck squamous cell carcinoma \\
KIRC & kidney renal clear cell carcinoma  \\
KIRP & kidney renal papillary cell carcinoma\\
LAML & acute myeloid leukemia \\
LGG & brain lower grade glioma\\
LIHC & liver hepatocellular carcinoma\\
LUAD & lung adenocarcinoma \\
LUSC & lung squamous cell carcinoma \\
OV & ovarian serous cystadenocarcinoma \\
PAAD & pancreatic adenocarcinoma\\
PCPG & pheochromocytoma and paraganglioma\\
READ & rectum adenocarcinoma \\
SARC & sarcoma\\
STAD & stomach adenocarcinoma\\
THCA & thyroid carcinoma\\
THYM & thymoma\\
UCEC & endometrial carcinoma \\
\\
%\hdashline
COADREAD & colon cancer + rectal cancer \\
\hline
\end{tabular}
\end{center}



\newpage

# Session info {.unnumbered}

```{r sessionInfo, echo=FALSE}
sessionInfo()
```