---
title: "Vignette Title"
author: "TBD"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

## Introduction to the (community-) AMARETTO algorithm and software toolbox :

Computational inference of regulatory networks underlying complex human diseases is one of the fundamental goals of systems biology and has shown great promise for deciphering the regulatory cell circuits driving complex disease biology, especially in cancer. The availability of increasing volumes of multimodal data ranging from multi-omics to imaging and clinical data across multiscale systems promises to improve our understanding of the regulatory mechanisms underlying complex human diseases. The main challenges are to integrate the multiple levels of multimodal data within biological systems and to translate them across multiscale biological systems to decipher the underpinnings of human diseases.

Here we introduce the (Community-)AMARETTO framework as a toolbox for learning how regulatory networks - cell circuits and their drivers - are shared or distinct within and across biological systems with a broad range of applications, from disease subtyping to driver and drug discovery in studies of human disease such as cancer. The (community) AMARETTO toolbox currently consists of two algorithmic tools:

-  The AMARETTO algorithm facilitates multimodal inference of regulatory networks within one biological system via multi-omics data fusion (e.g., genetic, epigenetic, transcriptomic, proteomic) and association with phenotypes derived from clinical (e.g., diagnostic and prognostic) and imaging (e.g., histopathology and radiographic) data.

-  The Community-AMARETTO algorithm enables multiscale inference to learn how these regulatory networks are shared or distinct across biological systems (e.g., across diseases, across cohorts, across model systems and patient studies, and across in vitro and in vivo systems).

The (Community-) AMARETTO framework is available as user-friendly tools from GitHub, Bioconductor, GenePattern, GenomeSpace, GenePattern Notebook and R Jupyter Notebook (see Resources).

Beyond our recent applications to studies of cancer (see References) the (Community-) AMARETTO software toolbox is more generally applicable to studies of human disease, including cancer, infectious, neurologic and immune-mediated diseases.


## *AMARETTO core tools and downstream analytic functionalities

The *AMARETTO framework currently consists of two algorithmic software tools. First, AMARETTO infers regulatory networks via multi-omics data fusion within each biological system. Specifically, AMARETTO identifies potential cancer drivers by identifying genes whose genetic and epigenetic cancer aberrations have a direct functional impact on their own transcriptomic or proteomic expression. These (epi)genetic drivers can be augmented, intersected or replaced with predefined candidate drivers with known regulatory function (e.g., transcription factors from TFutils). AMARETTO then connects these drivers in a regulatory program with modules of co-expressed target genes that they putatively control, defined as regulatory modules or cell circuits, using a regularized regression algorithm (i.e., Elastic Net regression). Next, Community-AMARETTO learns communities or subnetworks by connecting regulatory networks inferred from different systems using an edge betweenness community detection algorithm (i.e., Girvan Newman) to identify cell circuits and drivers that are shared and distinct across biological systems and diseases.

The *AMARETTO framework additionally offers tools for downstream analytic functionalities on both module and community levels, including functional annotation of modules and communities (e.g., using known functional categories from MSigDB), stratifying modules and communities for increasingly specific phenotypes (e.g., patient characteristics such as survival, molecular subclasses, known (epi)genetic cancer aberrations, or features derived from histopathology and radiographic imaging, as well as in-depth studies of etiologies of cancer via spatiotemporal - time course and/or single-cell - studies in model systems), validation of predicted drivers (e.g., using genetic perturbation studies in model systems â€“ knockdown or overexpression experiments of driver genes), discovering drugs targeting drivers and their predicted target genes (e.g., using chemical perturbation studies in model systems), and systematic assessment and benchmarking of the networks for generalized prediction performance of the (sub)networks.

#### In this tutorial, we will guide you through the following steps for running the *AMARETTO toolbox in R via GitHub and Bioconductor

Step 1: Before you begin, install software for running *AMARETTO
Step 2: Preparing data for running *AMARETTO
Step 3: Running AMARETTO on first example study: inferring regulatory networks via multi-omics data fusion for TCGA LIHC patient cohort
Step 4: Running AMARETTO on second example study: inferring regulatory networks from RNA-Seq data from CCLE liver cell lines
Step 5: Running Community-AMARETTO to combine both example studies: identifying regulatory subnetworks or communities shared and distinct across TCGA and CCLE cohorts

##Step 1: Before you begin, install software for running *AMARETTO
Import libraries required for running the *AMARETTO in R Notebook by running the next code cells.

The *AMARETTO toolbox can be installed from GitHub (development versions) or from Bioconductor (official releases).

For GitHub repositories you can proceed with Step 1.a.

For Bioconductor repositories you can proceed with Step 1.b.


Step 1.a. Installing the *AMARETTO toolbox from GitHub (Development Versions)

Installing and loading AMARETTO

Installing the latest versions of AMARETTO. This may take a few minutes...

```{r}
library(devtools)
install_github("broadinstitute/CommunityAMARETTO",ref='develop',force = 'TRUE')
library("CommunityAMARETTO")
```

```{r}
devtools::install_github("gevaertlab/AMARETTO",ref="for35_develop",quiet = FALSE)
library("AMARETTO")
```


Step 3.a. Preparing data and parameter settings for running AMARETTO

Loading multi-omics data from TCGA LIHC cohort

Loading Gene Expression (MA) data from TCGA LIHC patient cohort (Required)
```{r}
data("ProcessedDataTCGA")
data("ProcessedDataCCLE")
data("Driver_Genes")
```

Loading Copy Number Variation (CNV) data from TCGA LIHC cohort (Optional)
```{r}
AMARETTOinit_TCGA <- AMARETTO::AMARETTO_Initialize(ProcessedData=ProcessedDataTCGA,
                                            Driver_list = Driver_Genes$TFs_TFutils_union,
                                            method = "union",
                                            NrCores = 1,
                                            NrModules = 10,
                                            VarPercentage = 10)


AMARETTOResults_TCGA<-AMARETTO::AMARETTO_Run(AMARETTOinit_TCGA)
```
```{r}
samples_TCGA<-readr::read_delim('./sample_info/Samples_LIHC_phenotypes.txt','\t')
phenotype_tests_TCGA<-readr::read_delim("./sample_info/Samples_LIHC_statistics.txt",delim="\t",col_names=TRUE)
```

```{r}
TCGA_phenotype_tests_all<-AMARETTO::AMARETTO_unite_results(AMARETTO::AMARETTO_statistical_test(AMARETTOinit_TCGA,
                                                                                               AMARETTOResults_TCGA,
                                                                                               samples_TCGA,
                                                                                               phenotype_tests_TCGA))
```

```{r}
samples_LIHC$SurvivalTime<-as.factor(cut(samples_LIHC$SurvivalTime, 7, include.lowest=TRUE, dig.lab = 5))
samples_LIHC$SurvivalCensoring<-as.factor(samples_LIHC$SurvivalCensoring)
```

```{r}
html_report_TCGA_address<-'html_report_TCGA_address'
hyper_geo_reference_address<-'../inst/extdata/h.all.v6.2.symbols.gmt'
AMARETTO_HTMLreport(AMARETTOinit_TCGA,
                    AMARETTOResults_TCGA,
                    ProcessedData = ProcessedDataTCGA,
                    hyper_geo_test_bool = TRUE,
                    hyper_geo_reference = hyper_geo_reference_address,
                    output_address = html_report_TCGA_address,
                    MSIGDB = TRUE,
                    SAMPLE_annotation = NULL)
```


```{r}
AMARETTOinit_CCLE <- AMARETTO::AMARETTO_Initialize(ProcessedData = ProcessedDataCCLE,
                                            Driver_list = Driver_Genes$TFs_TFutils_union,
                                            method = "union",
                                            NrCores = 1,
                                            NrModules = 10,
                                            VarPercentage = 10)
AMARETTOResults_CCLE<-AMARETTO::AMARETTO_Run(AMARETTOinit_CCLE)
```

```{r}



```

```{r}
html_report_CCLE_address<-'html_report_CCLE_address'
hyper_geo_reference_address<-'../inst/templates/h.all.v6.2.symbols.gmt'
AMARETTO_HTMLreport(AMARETTOinit_CCLE,
                    AMARETTOResults_CCLE,
                    ProcessedData = ProcessedDataCCLE,
                    hyper_geo_test_bool = TRUE,
                    hyper_geo_reference = hyper_geo_reference_address,
                    output_address = html_report_CCLE_address,
                    MSIGDB = TRUE,
                    SAMPLE_annotation = NULL)

```




##Step 3.b. Running AMARETTO via multi-omics data fusion

Running AMARETTO for inferring regulatory networks from multi-omics data
By running the following code cell, you can now run the AMARETTO algorithm.



## Step 4. Running AMARETTO on second example study: inferring regulatory networks from RNA-Seq data from CCLE liver cell lines
The AMARETTO algorithm that infers regulatory networks within one cohort or biological system can be run in multiple ways: (1) via multi-omics data fusion in case genetic, epigenetic and functional genomics data are available (see example in previous Step 3 for multi-omics data from TCGA), or (2) if only functional genomics data are available (see example in this Step 4 for transcriptomic data from CCLE). See Step 3 for more detailed information.

Loading Gene Expression (MA) data from CCLE liver cell lines (Required)
Defining List(s) of Candidate Driver Genes (Required)
In this example, we precompiled a list of candidate driver genes that takes the union of TCGA genetic and epigenetic drivers with TFutils transcription factors (same list of candidate drivers as in Step 3)

##Setting parameters for running AMARETTO (Required)
Core parameters that can be set by the user for running AMARETTO. See Step 3 for more detailed information.


Setting parameters for generating HTML results reports (Optional)
Additional parameters that can be set by the user for running AMARETTO. See Step 3 for more detailed information.

tting parameters for computing on servers (Optional)
Selecting the number of cores for parallel computing.


##Step 4.b. Running AMARETTO on transcriptomics data

Running AMARETTO for inferring regulatory networks from transcriptomics data
By running the following code cell, you can now run the AMARETTO algorithm.

##Step 5. Running Community-AMARETTO to combine both example studies: identifying regulatory subnetworks or communities shared and distinct across TCGA and CCLE cohorts


Loading two or more results from AMARETTO, in this example the previous TCGA and CCLE results (Required)
Selecting AMARETTO analyses for Community-AMARETTO analysis. The user can submit the .rds files that represent the corresponding results from at least two or more previous AMARETTO analyses (see above, run in Steps 3 and 4).

```{r}
AMARETTOinit_TCGA <- readRDS(file="TCGA_AMARETTOinit.rds")
AMARETTOresults_TCGA <- readRDS(file="TCGA_AMARETTOresults.rds")

AMARETTOinit_CCLE <- readRDS(file="CCLE_AMARETTOinit.rds")
AMARETTOresults_CCLE <- readRDS(file="CCLE_AMARETTOresults.rds")

HTMLsAMARETTOlist <- c("TCGA"=output_directory_TCGA,"CCLE"=output_directory_CCLE)
```

Loading additional networks as a set of signatures in .GMT format (Optional)

One or more additional networks can be submitted as signatures files in .GMT format and combined by running the following code cell. These networks are used in Community-AMARETTO as separate networks. In this example, we submit previously published signatures and/or networks, such as immune signatures from CiberSort, stemness signatures from Ben-Porath et al., and diagnostic and prognistic liver cancer signatures from Hoshida et al. as three independent networks to be analyzed together with the liver cancer networks derived from TCGA in Step 3 and CCLE in Step 4.

If additional networks are submitted, please run following cell code to include them in the analysis.


```{r}
ImmuneSignatures <- "ImmuneSignatures.gmt"
download.file(url="https://www.broadinstitute.org/~npochet/NotebookExample/ExampleData/ImmuneSignatures.gmt", destfile=ImmuneSignatures)

StemSignatures <- "StemSignatures.gmt"
download.file(url="https://www.broadinstitute.org/~npochet/NotebookExample/ExampleData/StemSignatures.gmt", destfile=StemSignatures)

LiverSignatures <- "LiverSignatures.gmt"
download.file(url="https://www.broadinstitute.org/~npochet/NotebookExample/ExampleData/LiverSignatures.gmt", destfile=LiverSignatures)

list_additional_networks = list(ImmuneSignatures = "ImmuneSignatures.gmt", StemSignatures = "StemSignatures.gmt", LiverSignatures = "LiverSignatures.gmt")
```

```{r}
list_additional_networks = NULL
```



Setting parameters for generating HTML results reports (Optional)
Additional parameters that can be set by the user for running Community-AMARETTO include:

(1) Define collections of known functional categories for functional characterization of the regulatory subnetworks or communities, such as, for example, the Hallmark and C2 collections from MSigDB in this example. More collections can be downloaded from http://software.broadinstitute.org/gsea/msigdb/collections.jsp) and/or uploaded by the user in .GMT format.

(2) Provide a base "file name" for output files generated by the Community-AMARETTO analysis (e.g., myCommunityAmarettoAnalysis).

```{r}
genesets_database_reference <- "H_C2_genesets.gmt"
download.file(url="https://www.broadinstitute.org/~npochet/NotebookExample/ExampleData/H_C2_genesets.gmt", destfile=genesets_database_reference) 

output_directory_cAMARETTO = "./cAMARETTO_report/"
dir.create(output_directory_cAMARETTO)
```


Setting parameters for computing on servers (Optional)
Selecting the number of cores for parallel computing.


```{r}
NrCores = 4
```

Step 5.b. Running Community-AMARETTO
By running the following code cells, you can now run the Community-AMARETTO algorithm.
You can run the following code cell for computing the extent of overlaps between all regulatory modules learned across data cohorts.
```{r}
cAMARETTOresults<-cAMARETTO_Results(AMARETTOinit_all = list(TCGA=AMARETTOinit_TCGA,CCLE=AMARETTOinit_CCLE) ,
                                    AMARETTOresults_all = list(TCGA=AMARETTOresults_TCGA,CCLE=AMARETTOresults_CCLE),
                                    gmt_filelist = list_additional_networks,
                                    NrCores = NrCores,
                                    drivers = FALSE)
```
                                    Creating and filtering edges in the module networks across data cohorts. Additional parameters that can be set by the user in this step include:

(1) Filtering edges in the subnetworks or communities based on p-value significance. Edges in the network with p-values larger than the cutoff value will be filtered out. The default cutoff p-value is 0.05.

(2) Filtering edges in the subnetworks or communities based on the minimum number overlapping genes. Edges in the network with the number of overlapping genes less than the cutoff value will be filtered out. The default cutoff value is 5 overlapping genes.

```{r}
cAMARETTOnetworkM<-cAMARETTO_ModuleNetwork(cAMARETTOresults,
                                           pvalue = 0.05,
                                           inter = 5)
```


Identifying and filtering edges in the subnetworks across data cohorts. Additional parameters that can be set by the user in this step include:

Filtering edges in the subnetworks or communities that do not satisfy all the following conditions:
(1) Number of nodes in the community larger than the 1% of the total number of nodes in the network,
(2) Number of represented datasets/cohorts in the community larger than the 10% of the subnetwork size (and at least, larger than 2),
(3) Ratio between edges inside/outside the community larger than 0.5.

The user can choose between filtering according to these criteria, in which case edges in the network that do not satisfy all of these criteria will be filtered out, or whether to not apply these filtering criteria to retain all edges.


```{r}
cAMARETTOnetworkC<-cAMARETTO_IdentifyCom(cAMARETTOnetworkM,
                                         filterComm = FALSE,
                                         ratioCommSize = 0.01,
                                         MinRuns = 2,
                                         ratioRunSize = 0.1,
                                         ratioEdgesInOut = 0.5)
```

Generating the Community-AMARETTO .html reports that directly links to the original AMARETTO .html reports.

```{r}
cAMARETTO_HTMLreport(cAMARETTOresults = cAMARETTOresults,
                     cAMARETTOnetworkM = cAMARETTOnetworkM,
                     cAMARETTOnetworkC = cAMARETTOnetworkC,
                     HTMLsAMARETTOlist = HTMLsAMARETTOlist,
                     CopyAMARETTOReport = TRUE,
                     hyper_geo_test_bool = TRUE,
                     hyper_geo_reference = genesets_database_reference,
                     MSIGDB = TRUE,
                     output_address= output_directory_cAMARETTO)

```

Step 5.c. Viewing Community-AMARETTO results combining multiple AMARETTO analyses
The .zip file generated in previous step provides a queryable .html Community-AMARETTO report that directly links to the original AMARETTO analyses from which they were derived. The Community-AMARETTO report includes:

(1) An overview index.html page that provides:

A summary of the analyzed networks (including links to the original AMARETTO networks derived from multiple datasets)
A network graph overview of the subnetworks or communities learned across multiple datasets
A queryable overview table of shared/distinct communities with assignments of modules to communities across all datasets
A queryable overview table of shared/distinct drivers of communities with assignments of drivers to communities across all datasets
A queryable overview table of gene assignments (drivers and targets) to communities
A queryable overview table with enrichments of known functional categories in communities
(2) For each regulatory subnetwork or community a community#.html page is generated that provides:

A network graph visualization of the subnetworks or communities learned across multiple datasets
A queryable table of shared/distinct communities across multiple datasets, including regulatory modules that are shared/distinct across datasets
A queryable table of gene assignments (drivers and targets) in communities
A queryable table with enrichments of known functional categories in communities
Overview and community-specific Tables and Network Graphs can be saved as .CSV, .Excel, .PDF and .PNG files.


Time complexity of *AMARETTO
Depending on the size of the data, for example, for TCGA cohorts of ~300-500 samples, it can take up to several hours to run the *AMARETTO algorithms and generate reports on the Google Colab servers.


Resources
The source code and user-friendly tools of the current *AMARETTO toolbox and future developments are available from GitHub, Bioconductor, GenePattern, GenomeSpace, GenePattern Notebook and R Jupyter Notebook.

*AMARETTO in Bioconductor
AMARETTO in Bioconductor: https://www.bioconductor.org/packages/devel/bioc/html/AMARETTO.html
Community-AMARETTO in Bioconductor: in preparation for submission
*AMARETTO in GitHub
AMARETTO in GitHub: https://github.com/gevaertlab/AMARETTO
Community-AMARETTO in GitHub: https://github.com/broadinstitute/CommunityAMARETTO
*AMARETTO in GenePattern
AMARETTO in GenePattern:
https://cloud.genepattern.org/gp/pages/index.jsf?lsid=urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00378
Community-AMARETTO in GenePattern:
https://cloud.genepattern.org/gp/pages/index.jsf?lsid=urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00381
*AMARETTO in GenomeSpace
The AMARETTO and Community-AMARETTO modules in GenePattern are also available within GenomeSpace: http://www.genomespace.org/

*AMARETTO Notebooks
The *AMARETTO in GenePattern and R Notebooks provide users with a complete analysis pipeline that enables running AMARETTO on one or multiple data cohorts and connecting them using Community-AMARETTO. Each AMARETTO and Community-AMARETTO analysis generates a detailed report of genome-wide networks inferred from one cohort and/or shared/distinct across multiple cohorts. These reports include queryable tables and visualizations (heatmaps and network graphs) of shared/distinct cell circuits and their drivers, as well as their functional and phenotypic characterizations.

*AMARETTO in GenePattern Notebook: GenePattern Notebook from https://notebook.genepattern.org/
*AMARETTO in R via GitHub and Bioconductor: Jupyter Notebook from https://colab.research.google.com/
*AMARETTO example reports
Studying hepatitis C & B virus-induced hepatocellular carcinoma using AMARETTO & Community-AMARETTO:

An example report that learns regulatory networks from multi-omics data for hepatocellular carcinoma based on integrating genetic, epigenetic and functional genomics data from TCGA: AMARETTO Report
An example report that integrates regulatory networks derived from >6 liver data sources (multi-omics hepatocellular carcinoma patient data from TCGA, ~25 liver cell line models from CCLE, time course hepatitis C virus infection data in Huh7 models, time course hepatitis B virus infection data in HepG2 models, single-cell hepatitis C virus infection data in Huh7 models, single-cell hepatitis B virus infection data in HepG2 models, further augmented with previously published prognostic network models that were derived from hepatocellular carcinoma patient data): Community-AMARETTO Report
An example of ongoing work on developing gene-level ontology network representations from AMARETTO modules & Community-AMARETTO communities: Shiny App
Multi-omics & imaging data fusion for glioblastoma multiforme using AMARETTO:

An example report that integrates imaging data into the multi-omics regulatory networks for glioblastoma multiforme based on multi-omics and non-invasive imaging data from TCGA/TCIA (that we will later connect with networks learned from integrating RNA-Seq refined for anatomic structures and stem cells with histopathology imaging data from IvyGAP and that we will subsequently further refine based on single-cell RNA-Seq studies): AMARETTO Report



#References
Multiscale and multimodal inference of regulatory networks using *AMARETTO. In preparation for submission.

Champion M., Brennan K., Croonenborghs T., Gentles A. J., Pochet N., Gevaert O. (2018). Module Analysis Captures Pancancer Genetically and Epigenetically Deregulated Cancer Driver Genes for Smoking and Antiviral Response. EBioMedicine, 27, 156-166. PMID:29331675 PMCID:PMC5828545

Gevaert O., Villalobos V., Sikic B. I., Plevritis S. K. (2014). Identification of ovarian cancer driver genes by using module network integration of multi-omics data. Interface Focus, 3(4), 20130013. PMID:24511378 PMCID:PMC3915833

Gevaert O., Tibshirani R., Plevritis S. K. (2015). Pancancer analysis of DNA methylation-driven genes using MethylMix. Genome Biology, 16(1), 17. PMID:25631659 PMCID:PMC4365533

Gevaert O. (2015). MethylMix: an R package for identifying DNA methylation-driven genes. Bioinformatics, 31(11), 1839-41. PMID:25609794 PMCID:PMC4443673

Stubbs B. J., Gopaulakrishnan S., Glass K., Pochet N., Everaert C., Raby B., Carey V. (2019). TFutils: Data structures for transcription factor bioinformatics. F1000Research, 8:152. (https://doi.org/10.12688/f1000research.17976.1)

Reich M., Liefeld T., Ocana M., Jang D., Bistline J., Robinson J., Carr P., Hill B., McLaughlin J., Pochet N., Borges-Rivera D., Tabor T., Thorvaldsdottir H., Regev A., Mesirov J. P. (2013). GenomeSpace: an environment for frictionless bioinformatics. F1000Posters, 4:804 (https://f1000research.com/posters/1093972)

Qu K., Garamszegi S., Wu F., Thorvaldsdottir H., Liefeld T., Ocana M., Borges-Rivera D., Pochet N., Robinson J. T., Demchak B., Hull T., Ben-Artzi G., Blankenberg D., Barber G. P., Lee B. T., Kuhn R. M., Nekrutenko A., Segal E., Ideker T., Reich M., Regev A., Chang H. Y., Mesirov J. P. (2016). Integrative genomic analysis by interoperation of bioinformatics tools in GenomeSpace. Nature Methods, 13(3), 245-247. PMID:26780094 PMCID:PMC4767623

Cedoz PL, Prunello M, Brennan K, Gevaert O. MethylMix 2.0: an R package for identifying DNA methylation genes. Bioinformatics. 2018 Sep 1;34(17):3044-3046. doi: 10.1093/bioinformatics/bty156. PubMed PMID: 29668835; PubMed Central PMCID: PMC6129298.

Gevaert O, Tibshirani R, Plevritis SK. Pancancer analysis of DNA methylation-driven genes using MethylMix. Genome Biology 2015 Jan 29;16:17. doi: 10.1186/s13059-014-0579-8. PubMed PMID: 25631659; PubMed Central PMCID: PMC4365533.

Gevaert O. MethylMix: an R package for identifying DNA methylation-driven genes. Bioinformatics. 2015 Jun 1;31(11):1839-41. doi: 10.1093/bioinformatics/btv020. Epub 2015 Jan 20. PubMed PMID: 25609794; PubMed Central PMCID: PMC4443673.



Funding
This work was supported by grants from NIH NCI ITCR R21 CA209940 (Pochet), NIH NCI ITCR U01 CA214846 Collaborative Supplement (Carey/Pochet) and NIH NIAID R03 AI131066 (Pochet).


Questions?
For any questions with the *AMARETTO Notebooks, please contact Nathalie Pochet (npochet@broadinstitute.org) and Olivier Gevaert (olivier.gevaert@stanford.edu).





